sdk/rust/src/ledger_id.rs

```diff
@@ -55,6 +69,66 @@ impl Debug for LedgerId {
 
 impl Display for LedgerId {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        f.pad(&base64::encode(&self.0))
+        if self.is_mainnet() {
+            f.write_str("mainnet")
+        } else if self.is_testnet() {
+            f.write_str("testnet")
+        } else if self.is_previewnet() {
+            f.write_str("previewnet")
+        } else {
+            f.write_str(&hex::encode(self.to_bytes()))
+        }
+    }
+}
+
+impl FromStr for LedgerId {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "mainnet" => Ok(Self::mainnet()),
+            "testnet" => Ok(Self::testnet()),
+            "previewnet" => Ok(Self::previewnet()),
+            _ => hex::decode(s).map(Self::from_bytes).map_err(Error::basic_parse),
```

Comment:
do you need to strip a `0x` prefix here?

Reply:
The java SDK does not.